llvm/lib/IR/Globals.cpp

```diff
@@ -335,6 +335,14 @@ bool GlobalObject::canIncreaseAlignment() const {
   if (isELF && !isDSOLocal())
     return false;
 
+  bool isXCOFF =
+      (!Parent || Triple(Parent->getTargetTriple()).isOSBinFormatXCOFF());
+  if (isXCOFF)
+    if (const GlobalVariable *GV = dyn_cast<GlobalVariable>(this))
+      // GV with toc-data attribute is defined in a TOC entry which has a fixed
```

Comment:
This comment can be lifted before `isXCOFF`. It seems not correct.

The alignment is not increased to mitigate TOC size overflow.